src/main/flight/pid.c

```diff
@@ -439,7 +441,22 @@ STATIC_UNIT_TESTED FAST_CODE_NOINLINE float pidLevel(int axis, const pidProfile_
     angle += gpsRescueAngle[axis] / 100; // ANGLE IS IN CENTIDEGREES
 #endif
     angle = constrainf(angle, -pidProfile->levelAngleLimit, pidProfile->levelAngleLimit);
-    const float errorAngle = angle - ((attitude.raw[axis] - angleTrim->raw[axis]) / 10.0f);
+    float rawangle = ((attitude.raw[axis] - angleTrim->raw[axis]) / 10.0f);
+    //if (isUpsidedown() && ((IS_RC_MODE_ACTIVE(BOX3D) || flight3DConfig()->switched_mode3d)) ){
+    if ( isUpsidedown() && featureIsEnabled(FEATURE_3D) && !IS_RC_MODE_ACTIVE(BOX3D) ){
+        if (axis == PITCH){
+            rawangle *=-1;
+        }
+            else{ //ROLL
+                if (rawangle < 0){
+                    rawangle = rawangle + 180;
+                }
+                else{
+                    rawangle = rawangle - 180;
+                }
+            }
```

Comment:
```suggestion
        switch (axis) {
        case PITCH:
            rawangle *= -1;
            break;
        case ROLL:
            rawangle -= SGN(rawangle) * 180;
            break;
        }
```

Reply:
I just completed a test flight with the updated switch operation and is working as designed.